src/scripts/generate-binary-ops.py

```diff
@@ -0,0 +1,318 @@
+        calc = random.randint(2, 3) # 2 or 3
+    elif arithcalc and urscalc and not bitcalc:
+        calc = (1 + random.randint(0, 1) * 2) # Must be 1 or 3
+    elif arithcalc and bitcalc and urscalc:
+            calc = random.randint(1, 3) # 1, 2 or 3
+    return calc
+
+# Setting maximum and minimum depending on operation type
+# 1: arithmetic, 2: bitwise (excluding unsigned right shift) 3: unsigned right shift
+def set_min_max(calc):
+    if calc == 1:
+        MINNUM = _constants.min
+        MAXNUM = _constants.max
+    elif calc == 2:
+        MINNUM = _constants.bitmin
+        MAXNUM = _constants.bitmax
+    else:
+        MINNUM = _constants.uint32min
+        MAXNUM = _constants.uint32max
+    return {'MIN' : MINNUM, 'MAX': MAXNUM}
+
+# Converting expression, expected value, false value in a function call to string format
+def string_creator(gs, options, MINNUM, MAXNUM):
+    last = gs.operands[0]
+    expression = ""
+    expression += ("%s%d" % ("(" * (gs.num_operands - 1), last))
+    for i in range(options.operand_count - 1):
+        result = gs.operators_list[gs.operators[i]](last, gs.operands[i + 1])
+        expression += (" %s %d)" % (result["oper"], result["op2"]))
+        if result["oper"] == "/":
+            expression += " | 0)"
+            expression = "(" + expression
+        gs.operands[i + 1] = result["op2"]
+        last = result["sum"]
+    expression += "\n"
+    expected_value = last
+
+    # Random number wich must not be equal with the expected value
+    if ((random.randint(0, 1) and last != MAXNUM) or last == MINNUM ):
+        N1 = random.randint(math.ceil(last + 1), MAXNUM)
+        N2 = random.randint(math.ceil(last + 1), MAXNUM)
+        N3 = random.randint(math.ceil(last + 1), MAXNUM)
+    else:
+        N1 = random.randint(MINNUM, math.floor(last - 1))
+        N2 = random.randint(MINNUM, math.floor(last - 1))
+        N3 = random.randint(MINNUM, math.floor(last - 1))
+    false_values = "[(%s), (%s), (%s)]" % (str(N1), str(N2), str(N3))
+    function_call = ("validate_numeric((%s), (%s), (%s));\n" % (expression, str(expected_value), false_values))
+    return function_call
+
+# Generator
+def main(options):
+    SEED = options.seed
+    random.seed(SEED)
+    generated_test_counter = 0 # How many tests are written (defending against script error: null)
+    file_count = 0 # How many files are created / written
+    arithcalc = False # If we need to do arithmetic calculations
+    bitcalc = False # If we need to do bitwise calculations
+    urscalc = False # If we need to do unsigned right shift (abbreviation: unsigned right shift calculation)
+    calc = 0 # Actually are we doing arithmetic (1) or bitwise (excluding unsigned right shift) (2) calculations
+                # or shift right zero (3) - No (0) is unsetted
+    arithcalc = needed_calculation_types(options)["arithcalc"]
+    bitcalc = needed_calculation_types(options)["bitcalc"]
+    urscalc = needed_calculation_types(options)["urscalc"]
```

Comment:
After this point, the following shouldn't the `file_writer` script's task? (except to generate the content)

Reply:
We respectfully disagree with this, because we use different methods for everything excepted the file writing. But I corrected and agreed with all the other reviews of yours.

Reply:
So you mean, you really need this whole file{writing, cutting, separating} part in every generator script??? The only difference between each script is the content. By the `file_writer` script I meant an universal script which gets a `relatively big {string, list, dictionary}` whatever is needed (thats what the generator scripts produce) and **THIS** script do all the job, what is needed to create the file(s) (cutting, separating).